### PR TITLE
Add systrace support on Apple platforms

### PR DESCRIPTION
**Pull request by bejado**
This add supports for `SYSTRACE` macros on Apple platforms. Systrace intervals are recorded as signposts and can be viewed in Instruments.

### CODE REVIEW

File: libs/utils/include/utils/darwin/Systrace.h [hunk 1]
```diff
@@ -0,0 +1,259 @@
...
+#define SYSTRACE_DISABLE()
+#define SYSTRACE_CONTEXT()
+#define SYSTRACE_NAME(name)
+#define SYSTRACE_NAME_BEGIN(name)
+#define SYSTRACE_NAME_END()
+#define SYSTRACE_CALL()
+#define SYSTRACE_ASYNC_BEGIN(name, cookie)
+#define SYSTRACE_ASYNC_END(name, cookie)
+#define SYSTRACE_VALUE32(name, val)
+#define SYSTRACE_VALUE64(name, val)
```
Comment (poweifeng) [inline libs/utils/include/utils/darwin/Systrace.h:257]: Seems like this is covered in `utils/Systrace.h`?

**Review by poweifeng — COMMENTED** [review]
There seems to be quite a bit of code duplication between android and apple. Any thoughts on  de-duping?  Maybe abstract interfaces and implementation per platform?

File: libs/utils/include/utils/darwin/Systrace.h (hunk 1 above)
Comment (bejado) [reply libs/utils/include/utils/darwin/Systrace.h:257]: Yes, good catch.

**Comment by bejado** [comment]
@poweifeng I considered that, but I think performance is a concern here so I'm not sure if we want to introduce virtual calls. I'm sure Mathias will have some thoughts.

**Comment by pixelflinger** [comment]
> @poweifeng I considered that, but I think performance is a concern here so I'm not sure if we want to introduce virtual calls. I'm sure Mathias will have some thoughts.

Yes, we want these to be usable in release builds.

**Review by pixelflinger — APPROVED** [review]

**Review by poweifeng — APPROVED** [review]

**Pull request merged**